rapids-cmake/cuda/init_architectures.cmake

```diff
@@ -79,21 +71,22 @@ function(rapids_cuda_init_architectures project_name)
   # If `CMAKE_CUDA_ARCHITECTURES` is not defined, build for all supported architectures. If
   # `CMAKE_CUDA_ARCHITECTURES` is set to an empty string (""), build for only the current
   # architecture. If `CMAKE_CUDA_ARCHITECTURES` is specified by the user, use user setting.
-  if(DEFINED ENV{CUDAARCHS} AND "$ENV{CUDAARCHS}" STREQUAL "ALL")
-    set(cuda_arch_mode "ALL")
+  if(DEFINED ENV{CUDAARCHS} AND ("$ENV{CUDAARCHS}" STREQUAL "RAPIDS" OR "$ENV{CUDAARCHS}" STREQUAL
+                                                                        "ALL"))
+    set(cuda_arch_mode "RAPIDS")
   elseif(DEFINED ENV{CUDAARCHS} AND "$ENV{CUDAARCHS}" STREQUAL "NATIVE")
     set(cuda_arch_mode "NATIVE")
-  elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "ALL")
-    set(cuda_arch_mode "ALL")
+  elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "RAPIDS" OR CMAKE_CUDA_ARCHITECTURES STREQUAL "ALL")
+    set(cuda_arch_mode "RAPIDS")
   elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "" OR CMAKE_CUDA_ARCHITECTURES STREQUAL "NATIVE")
     set(cuda_arch_mode "NATIVE")
   elseif(NOT (DEFINED ENV{CUDAARCHS} OR DEFINED CMAKE_CUDA_ARCHITECTURES))
-    set(cuda_arch_mode "ALL")
+    set(cuda_arch_mode "RAPIDS")
   endif()
 
   # This needs to be run before enabling the CUDA language since RAPIDS supports the magic string of
   # "ALL"
```

Comment:
This comment seems inaccurate now -- but I'm not entirely sure about the purpose of this PR. The PR description is empty.

Reply:
Sorry about that. I have updated the description of the PR to explain what the end goal is. But in short we are moving away from the `ALL` keyword as the term is loaded now that `all` is handled by CMake / nvcc and produces a different set of cuda arch values.

The comment is stale now, as it should state that RAPIDS supports the magic keywords of `ALL` and `RAPIDS` ( both map to a `cuda_arch_mode` value of `RAPIDS` ). I have pushed an update to fix the comment